### ISSUE TITLE
Return error code "1295" for unsupported command in prepared statement

### ISSUE THREAD

**Issue by hebestreit** [StarRocks/starrocks#41757]
## Enhancement

We're trying to integrate Hashicorp Vault with StarRocks to generate database credentials using the [database secrets engine](https://developer.hashicorp.com/vault/docs/secrets/databases) and store them securely in Vault. Vault supports various databases and also [MySQL](https://developer.hashicorp.com/vault/docs/secrets/databases/mysql-maria).

After we've successfully connected Vault with StarRocks we stumbled over the problem that `ALTER USER` is an invalid statement type for prepared statements. From your documentation only SELECT statements are supported.
https://docs.starrocks.io/docs/sql-reference/sql-statements/prepared_statement/#prepare

```
2024-02-23 17:05:10,295 WARN (starrocks-mysql-nio-pool-30|9625) [ConnectProcessor.handleQuery():411] Failed to parse SQL: ALTER USER 'root'@'%' IDENTIFIED BY 'top-secret', because.
com.starrocks.common.AnalysisException: prepare statement only support QueryStatement
	at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:378) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:588) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:872) ~[starrocks-fe.jar:?]
	at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
```

Vault has implemented a fallback to run the statement without preparation when the command is not supported but only in case the returned error code is `1295`.
https://github.com/hashicorp/vault/blob/e6d2caaa52c71c6de0df76e183563e09bb0e65d2/plugins/database/mysql/mysql.go#L265-L282

The error code `1295` seems to be the correct one for unsupported commands in MySQL and it would be great if this can be also returned by StarRocks.

## Version

```
mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| 3.2.3-a40e2f8     |
+-------------------+
1 row in set (0.02 sec)
```

**Comment by hebestreit** [comment]
First of all thanks for implementing this change into StarRocks which will make the management of credentials so much easier.  

I've updated to 3.2.6 but it doesn't seem to work. Are the changes included in the latest 3.2.6 release? 

From the Git tag it seems like but unfortunately I'm still seeing the old behavior and getting back the error code 1064.

https://github.com/StarRocks/starrocks/blob/3.2.6/fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java#L121


Vault logs:

```
{... error":"1 error occurred:\n\t* failed to update user: failed to change password: Error 1064 (HY000): prepare statement only support QueryStatement\n\n"}
```
FE logs:
```
2024-04-23 15:07:30.319+02:00 WARN (starrocks-mysql-nio-pool-4|175) [ConnectProcessor.handleQuery():430] Failed to parse SQL: ALTER USER 'root'@'%' IDENTIFIED BY 'XXXXX', because.                                                                                                
test-starrocks-fe-0 com.starrocks.common.AnalysisException: prepare statement only support QueryStatement                                                  
test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:397) ~[starrocks-fe.jar:?]                                  
test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:607) ~[starrocks-fe.jar:?]                                     
test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:901) ~[starrocks-fe.jar:?]                                  
test-starrocks-fe-0     at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]                           
test-starrocks-fe-0     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]                                          
test-starrocks-fe-0     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]                                          
test-starrocks-fe-0     at java.lang.Thread.run(Thread.java:829) ~[?:?]      
```

I'm  running StarRocks in Kubernetes using the [starrocks-kubernetes-operator](https://github.com/StarRocks/starrocks-kubernetes-operator):

```
root@test-starrocks-fe-0:/opt/starrocks# ./fe/bin/show_fe_version.sh 
Build version: 3.2.6
Commit hash: 2585333
Build type: RELEASE
Build time: 2024-04-18 13:27:22
Build distributor id: ubuntu
Build user: StarRocks@localhost (Ubuntu 22.04.3 LTS)
Java compile version: openjdk full version "11.0.20+8-post-Ubuntu-1ubuntu122.04"
```

```
mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| 3.2.6-2585333     |
+-------------------+
1 row in set (0.03 sec)
```

**Comment by mofeiatwork** [comment]
> First of all thanks for implementing this change into StarRocks which will make the management of credentials so much easier.
> 
> I've updated to 3.2.6 but it doesn't seem to work. Are the changes included in the latest 3.2.6 release?
> 
> From the Git tag it seems like but unfortunately I'm still seeing the old behavior and getting back the error code 1064.
> 
> https://github.com/StarRocks/starrocks/blob/3.2.6/fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java#L121
> 
> Vault logs:
> 
> ```
> {... error":"1 error occurred:\n\t* failed to update user: failed to change password: Error 1064 (HY000): prepare statement only support QueryStatement\n\n"}
> ```
> 
> FE logs:
> 
> ```
> 2024-04-23 15:07:30.319+02:00 WARN (starrocks-mysql-nio-pool-4|175) [ConnectProcessor.handleQuery():430] Failed to parse SQL: ALTER USER 'root'@'%' IDENTIFIED BY 'XXXXX', because.                                                                                                
> test-starrocks-fe-0 com.starrocks.common.AnalysisException: prepare statement only support QueryStatement                                                  
> test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:397) ~[starrocks-fe.jar:?]                                  
> test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:607) ~[starrocks-fe.jar:?]                                     
> test-starrocks-fe-0     at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:901) ~[starrocks-fe.jar:?]                                  
> test-starrocks-fe-0     at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]                           
> test-starrocks-fe-0     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]                                          
> test-starrocks-fe-0     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]                                          
> test-starrocks-fe-0     at java.lang.Thread.run(Thread.java:829) ~[?:?]      
> ```
> 
> I'm running StarRocks in Kubernetes using the [starrocks-kubernetes-operator](https://github.com/StarRocks/starrocks-kubernetes-operator):
> 
> ```
> root@test-starrocks-fe-0:/opt/starrocks# ./fe/bin/show_fe_version.sh 
> Build version: 3.2.6
> Commit hash: 2585333
> Build type: RELEASE
> Build time: 2024-04-18 13:27:22
> Build distributor id: ubuntu
> Build user: StarRocks@localhost (Ubuntu 22.04.3 LTS)
> Java compile version: openjdk full version "11.0.20+8-post-Ubuntu-1ubuntu122.04"
> ```
> 
> ```
> mysql> select current_version();
> +-------------------+
> | current_version() |
> +-------------------+
> | 3.2.6-2585333     |
> +-------------------+
> 1 row in set (0.03 sec)
> ```

Sorry about that, `ALTER USER` doesn't return the expected error code, will fix it again.
